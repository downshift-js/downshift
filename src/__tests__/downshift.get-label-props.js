import React from 'react'
import {mount} from 'enzyme'
import Downshift from '../'

test('label "for" attribute is set to the input "id" attribute', () => {
  const wrapper = mount(<BasicDownshift />)
  const label = wrapper.find('label').first()
  const input = wrapper.find('input').first()
  expect(label.node.getAttribute('for')).toBeDefined()
  expect(label.node.getAttribute('for')).toBe(input.node.getAttribute('id'))
})

test('when the input id is set, the label for is set to it', () => {
  const id = 'foo'
  const wrapper = mount(<BasicDownshift inputProps={{id}} />)
  const label = wrapper.find('label').first()
  const input = wrapper.find('input').first()
  expect(label.node.getAttribute('for')).toBe(id)
  expect(label.node.getAttribute('for')).toBe(input.node.getAttribute('id'))
})

test('when the input id is set, and the label for is set to something else, an error is thrown', () => {
  expect(() =>
    mount(
      <BasicDownshift inputProps={{id: 'foo'}} labelProps={{htmlFor: 'bar'}} />,
    ),
  ).toThrowErrorMatchingSnapshot()
})

test('when the label for is set, and the input id is set to something else, an error is thrown', () => {
  expect(() =>
    mount(
      <BasicDownshift
        getLabelPropsFirst
        inputProps={{id: 'foo'}}
        labelProps={{htmlFor: 'bar'}}
      />,
    ),
  ).toThrowErrorMatchingSnapshot()
})

function BasicDownshift({
  inputProps,
  labelProps,
  getLabelPropsFirst = false,
  ...rest
}) {
  return (
    <Downshift {...rest}>
      {({getInputProps, getLabelProps}) => {
        if (getLabelPropsFirst) {
          labelProps = getLabelProps(labelProps)
          inputProps = getInputProps(inputProps)
        } else {
          inputProps = getInputProps(inputProps)
          labelProps = getLabelProps(labelProps)
        }
        return (
          <div>
            <input {...inputProps} />
            <label {...labelProps} />
          </div>
        )
      }}
    </Downshift>
  )
}
