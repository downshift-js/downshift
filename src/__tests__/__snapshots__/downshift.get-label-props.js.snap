// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the input id is set, and the label for is set to something else, an error is thrown 1`] = `"downshift: You provided the htmlFor of \\"bar\\" for your label, but the id of your input is \\"foo\\". You must either remove the id from your input or set the htmlFor of the label equal to the input id."`;

exports[`when the label for is set, and the input id is set to something else, an error is thrown 1`] = `"downshift: You provided the id of \\"foo\\" for your input, but the htmlFor of your label is \\"bar\\". You must either remove the id from your input or set the htmlFor of the label equal to the input id."`;
